redbot/setup.py

```diff
@@ -30,7 +31,7 @@
     config_dir.mkdir(parents=True, exist_ok=True)
 except PermissionError:
     print("You don't have permission to write to '{}'\nExiting...".format(config_dir))
-    sys.exit(1)
+    sys.exit(ExitCodes.CRITICAL)
```

Comment:
This should probably be `CONFIGURATION_ERROR`, since restarting won't give permission to write to this dir.

Reply:
Yeah, probably makes more sense. I'm guessing I *probably* used `CRITICAL` because it concerns `redbot-setup` which isn't a daemon process (unlike `redbot`) anyway and you shouldn't restart it after failure.

It's also possible that I didn't want to use `CONFIGURATION_ERROR` in `redbot-setup` at all, maybe because places where it could make sense don't concern Red's configuration. I don't know...

I think `CONFIGURATION_ERROR` probably makes more sense here, I'll make the change.